cranelift/codegen/src/isa/x64/encoding/rex.rs

```diff
@@ -298,16 +298,29 @@ pub(crate) fn emit_std_enc_mem(
 
     prefixes.emit(sink);
 
+    let mem_e = match mem_e.clone() {
+        Amode::RbpOffset { simm32, flags } => {
```

Comment:
This is to avoid providing the base reg as an operand to the operand collector, I guess?

I wonder if it might be cleaner, or less error-prone at least, to handle this case in the `Amode::get_operands` and `Amode::with_allocs` methods instead: skip providing rbp (or rsp) as an operand, and skip pulling an alloc from the AllocationConsumer if the original register was rbp (or rsp).

(I'm actually not sure; I'm going back and forth right now.)

The general concern I had with this is that we may not handle the same lowering everywhere. But actually I think everything funnels to this one lowering helper so maybe this is sufficient.

If we do stick to this approach, let's add a comment where we define the `RbpOffset` arm that it must be used for rbp (as opposed to the generic forms) because rbp cannot be given as an arg to regalloc.

Reply:
Yeah I actually like this better.